src/main/java/com/amazon/opendistroforelasticsearch/security/support/wildcard/Wildcard.java

```diff
@@ -0,0 +1,70 @@
+package com.amazon.opendistroforelasticsearch.security.support.wildcard;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+// Case-sensitivity and other options are defined by concrete Wildcard implementation
+public interface Wildcard {
```

Comment:
I don't think it is necessary to introduce `Wildcard` interface. Combine `Wildcard` with `WildcardMatcher` class, especially that `WildcardMatcher` is a more suitable name, IMO. Make `WildcardMatcher` abstract class that implements `Predicate`. All other `Wildcard` classes should be inner `private static` classes of `WildcardMatcher` class.

Reply:
Yeah, Wildcard -> WildcardMatcher is good idea.

Reply:
Done